### PR TITLE
Math

### PR DESCRIPTION
**Pull request by Virnkord**
5.7 Math functions.

### CODE REVIEW

File: test/cases/5.7_math.yaml [hunk 1]
```diff
@@ -31,11 +103,86 @@ tests:
...
+    expression: Math.n2.power(8)
+    result: []
+  - desc: '** Empty result when raising collection to the empty power'
+    expression: Math.n3.power(n2)
+    result: []
+  - desc: '** Error raising to a power if the input collection contains multiple items'
+    expression: Math.arr.power(8)
+    error: true
+  - desc: '** Error raising to a power due to too many input parameters'
+    expression: Math.n3.power([3, 5])
```
Comment (plynchnlm) [inline test/cases/5.7_math.yaml]: Needs a test for negative numbers:
(-1).power(0.5)
Also, please add (-1).power(1.5)

Comment (plynchnlm) [reply test/cases/5.7_math.yaml]: Note that it is not necessary to start all expressions with a resource name (Math).  You can just start with (-1)....  An advantage is that you don't have to look at the bottom of the file to see what the expression is operating on.

File: src/math.js [hunk 2]
```diff
@@ -65,6 +65,46 @@ engine.mod = function(x, y){
   return x % y;
 };
 
+engine.abs = function(x){
+  if (isEmpty(x)){
+    return [];
+  }else{
+    let num = ensureNumberSingleton(x);
+    if (typeof num != 'number'){
+      return num;
```
[the remark this answers is not in the document]
Comment (Virnkord) [reply src/math.js]: "When taking the absolute value of a quantity, the unit is unchanged."
(http://hl7.org/fhirpath/2019May/index.html#abs-integer-decimal-quantity)
Quantity as an array with a single string element. The only way to recognize quantity is to implement toQuantity() function (http://hl7.org/fhirpath/2019May/index.html#toquantityunit-string-quantity).
So, should I implement toQuantity() in misc.js and then import it to math.js?

File: test/cases/5.7_math.yaml (hunk 1 above)
Comment (Virnkord) [reply test/cases/5.7_math.yaml]: I will directly use the values instead of variables.
But if I start expression with negative number ((-1).power(1.5)), it throws an error "No PolarityExpression evaluator".

File: src/math.js [hunk 3]
```diff
@@ -65,6 +65,46 @@ engine.mod = function(x, y){
   return x % y;
 };
 
+engine.abs = function(x){
+  if (isEmpty(x)){
+    return [];
+  }else{
+    let num = ensureNumberSingleton(x);
+    if (typeof num != 'number'){
```
[the remark this answers is not in the document]
Comment (Virnkord) [reply src/math.js]: Fixed ensureNumberSingleton.

File: src/math.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Virnkord) [reply src/math.js]: I think abs() function should look something like that:

```
engine.abs = function(x){
  if (isEmpty(x)){
    return [];
  }else{
    if(x.toQuantity()){
      return x;
    }
    let num = ensureNumberSingleton(x);
    return Math.abs(num);
  }
};
```

[the remark this answers is not in the document]
Comment (Virnkord) [reply src/math.js]: Also, all of the string data would be recognized as quantity if they are convertable to quantity and I don't know how to set a quantity variable in my tests.

[the remark this answers is not in the document]
Comment (plynchnlm) [reply src/math.js]: > So, should I implement convertsToQuantity() in misc.js and then import it to math.js?
No, because I am working on adding Quantity (to add support for adding quantities to date/time).  So, hold off on anything involving Quantity; you might leave yourself a "TBD" comment to add code for those after my changes are in.

File: test/cases/5.7_math.yaml (hunk 1 above)
Comment (plynchnlm) [reply test/cases/5.7_math.yaml]: Right.  See pull request #13, which implements PolarityExpression.  For now, maybe for negative numbers just leave them in the resource.

File: src/math.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (plynchnlm) [reply src/math.js]: I will see if I can separate my PR so that the Quanitity type is defined in one PR, and the date/time + quantity stuff (which might take me a while) is in another.

**Comment by plynchnlm** [comment]
It is ready but just needs an update to CHANGELOG.md and the version number in package.json.
